pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -1775,6 +1776,102 @@ func ensurePVCCreated(t *testing.T, clients test.Clients, name, namespace string
 	}
 }
 
+// TestReconcileWithAffinityAssistantStatefulSet tests that given a pipelineRun with a PVC workspace,
+// an Affinity Assistant StatefulSet is created and that the Affinity Assistant name is propagated to TaskRuns.
+func TestReconcileWithAffinityAssistantStatefulSet(t *testing.T) {
```

Comment:
Would it make sense to test that if an assistant already exists, no new one is created?
Should it we test what happens with non-PVC backed workspaces?
It would be nice to see the case of two groups of tasks with two different PVC - so I would expect two different assistants. I know that we do not recommend that setup, but unless we explicitly block it, we need to support it.

Reply:
>Would it make sense to test that if an assistant already exists, no new one is created?

They would have _identical_ names, so Kubernetes would return `AlreadyExists` error, and that case is handled in the create-function.

>Should it we test what happens with non-PVC backed workspaces?

Yes, this is a good suggestion. I had two tasks, one with PVC workspace and one without workspace. But I have changed to three tasks now, using two different PVC workspaces, and the third using an emptyDir-workspace. This should cover the cases you describe and test-coverage is up.

>It would be nice to see the case of two groups of tasks with two different PVC - so I would expect two different assistants.

Yes, I added, so this case is covered now. Thanks.

>I know that we do not recommend that setup, but unless we explicitly block it, we need to support it.

This is actually a valid case. A `PipelineRun` can have multiple PVC-workspaces. But a single `Task` should not use more than one. E.g. you can have a Pipeline with two parallel "tracks" that builds two artifacts using two PVCs. Then a common Task can use the Task Results from both tracks and e.g. deploy both Image-Digests. That should work perfectly fine. The Tasks using different PVCs would execute on two separate Nodes (or the same) in a Best Effort fashion.